server/driver_tidb.go

```diff
@@ -391,7 +391,7 @@ func convertColumnInfo(fld *ast.ResultField) (ci *ColumnInfo) {
 			// Consider the decimal point.
 			ci.ColumnLength++
 		}
-	} else if fld.Column.Tp != mysql.TypeBit && fld.Column.Tp != mysql.TypeTiny {
+	} else if types.IsString(fld.Column.Tp) {
```

Comment:
My original question still remains open. Why does `types.IsString(fld.Column.Tp)` work for `year`? I understood this is used for solving the compatibility issue we found in Navicat. 
At least, it is worth a comment here.

Reply:
I think this is already explained in the PR description, we don't need to explain why old code causes some bug, because old code is removed.

Reply:
Firstly, `year` is a number typer, so it will not multiply `mysql.MaxBytesOfCharacter`.  
Secondly, using `types.IsString(fld.Column.Tp)` rather than just `fld.Column.Tp != mysql.TypeYear` can fix some other problem cause by this if branch.

Reply:
@zhexuany , For example, 

    create table t (k smallint(10) unsigned zerofill);
    insert into t value (10);

using jdbc test code above, TiDB and MySQL has different result.
MySQL: `0000000010`
TiDB: `0000000000000000000000000000000000000010`

This mean number type meet `zerofill` will has this problem. So we should use `types.IsString(fld.Column.Tp)`.

Reply:
> we don't need to explain why old code causes some bug, because old code is removed.

But we need why the fixes work. Information still resides at PR not code itself. 

When people read this code, they probably have no idea about this change. They have to go through PR which is a form of cost. I believe information has to come with the code. That is why we need comment.

Reply:
I will add some comment.

Reply:
Yes. We only need check string type simply because string type has truncation problem. This comment can provide context which can help other people better understand.